### PR TITLE
Support qadd_relu on pytorch mobile

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26992 Unify quantized conv and linear tests
* **#26982 Support qadd_relu on pytorch mobile**

Summary:
Fused add+relu support
Test Plan:
python test/test_quantized.py TestQNNPackOps.test_qnnpack_add

Also,
Add torch.backends.quantized.engine = "qnnpack"
Ran
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_different_qparams
python test/test_quantized.py TestQuantizedOps.test_qadd_relu_same_qparams
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D17635063](https://our.internmc.facebook.com/intern/diff/D17635063)

### CODE REVIEW

**Comment by jerryzh168** [comment]
No changes to the test?

**Comment by supriyar** [comment]
> No changes to the test?

Forgot to add. 
Eventually we need to merge the tests, but that can be a different PR.

**Review by jerryzh168 — APPROVED** [review]
LGTM

File: test/test_quantized.py [hunk 1]
```diff
@@ -1858,6 +1858,13 @@ def test_qnnpack_add(self, A, zero_point, scale_A, scale_B, scale_C):
             np.testing.assert_equal(qC, qC_qnnp.int_repr(),
                                     "Quantized addition failed.")
 
+            Crelu = C.copy()
+            Crelu[C < 0] = 0
+            qCrelu = _quantize(Crelu, scale_C, zero_point_C)
```
Comment (jerryzh168) [inline test/test_quantized.py]: I think we should just use `torch.quantize_per_tensor` instead of using `numpy`

**Pull request closed without merging**